### ISSUE TITLE
Reverse proxy with Nginx, but with a non-standard port on the reverse proxy

### ISSUE THREAD

**Issue by beerendlauwers** [firefly-iii/firefly-iii#3170]
I'm running Firefly III via Docker Compose and so I'd like to access it with a URL such as "sub.domain.com:5555" through a reverse proxy that is running on another machine. Making Firefly amenable to the proxy was pretty easy with `TRUSTED_PROXIES=**`, but when I access it via the domain, all the links in the HTML point to `https://sub.domain.com`, so without the non-standard port.

I've rooted around shortly in the source code, but I can only find the override for the `APP_URL`, which is command-line only, anyway. Is there another way for me to set a port number that gets slapped on at the end of the domain?

The weird thing is that a non-standard port does get detected if I don't go through the reverse proxy. I've changed the port mapping in the `docker-compose.yml` file from the regular `80:80` to something like `5555:80`, and then the port does show up in the HTML when I access it using my internal network's IP address.

Updating the reverse proxy to point to that port and going through the proxy with `https://sub.domain.com:5555` doesn't show the port number, either.

Has anyone managed to get something like this going?

**Comment by JC5** [comment]
This should be no different from other nginx configuration. Check out the nginx config in #3168

**Comment by beerendlauwers** [comment]
Found it. My Nginx reverse proxy was also running in Docker, and was portmapped to something like `5555:443`. Inside the config, I had the line

```
proxy_set_header X-Forwarded-Port       $server_port;
```

which resolved to `443`, of course. Because that's the default port number for secure HTTP connections, that port number wasn't showing up. I replaced`$server_port` with the external-facing port number and it works now.
